### PR TITLE
Fixes for cases where the neighboring lists are relatives

### PR DESCRIPTION
**Pull request by caffodian**
The previous change only made sense when two lists were beside each other.

In cases where you have say, a nested list beside another nested list, or just a nested list with siblings of the parent "beside" it, content destruction would occur.

### CODE REVIEW

File: plugins/list/plugin.js [hunk 1]
```diff
@@ -1090,12 +1090,15 @@
 								// if we have two neighboring lists, merge the list items from the second
 								// into the first, rather than moving one item at a time.
 								var startItem = nextLine.startContainer;
+								var commonAncestor = li.getCommonAncestor(startItem);
 								if (
 									startItem.type == CKEDITOR.NODE_ELEMENT &&
 									startItem.getName() == 'li' &&
 									startItem.getParent().getName() in listNodeNames &&
-									!li.getParent().contains(startItem)
```
Comment (caffodian) [inline plugins/list/plugin.js:1097]: This only checked if we were deleting two siblings into each other, and would skip our list-merging logic.

File: plugins/list/plugin.js [hunk 2]
```diff
@@ -1090,12 +1090,15 @@
...
+									!( commonAncestor.getName() in listNodeNames ) &&
```
Comment (caffodian) [inline plugins/list/plugin.js:1098]: superset of the previous case (if the list item we are currently in + the block being deleted into it have a common list ancesotr)

File: plugins/list/plugin.js [hunk 3]
```diff
@@ -1090,12 +1090,15 @@
...
+									!( li.contains( startItem ) )
```
Comment (caffodian) [inline plugins/list/plugin.js:1099]: also don't merge lists if the item being merged is already inside the list item are deleting into.

File: tests/plugins/list/backspace.html [hunk 4]
```diff
@@ -871,3 +871,53 @@
...
+	=>
+	<ul>
+		<li>a
+			<ul>
+				<li>^b</li>
+			</ul>
+		</li>
+		<li>c</li>
+	</ul>
+</textarea>
```
Comment (caffodian) [inline tests/plugins/list/backspace.html:894]: ^ is the caret position
and the => reflects the before/after
(in this case, pressing del)

Comment (jlward) [inline tests/plugins/list/backspace.html:894]: I assume this is some sort of test magic they built?

**Comment by jlward** [comment]
This looks good.

File: tests/plugins/list/backspace.html (hunk 4 above)
Comment (caffodian) [inline tests/plugins/list/backspace.html:894]: Yea, they use a similar pattern for most of the DOM manipulation cases.  Their testing framework has pretty good tools for comparing before/after HTML

**Pull request merged**